R/plot_taxa.R

```diff
@@ -239,35 +470,101 @@ plot_taxa_diversity <- function(observation, id, alpha = 1) {
 #' @export
 #' 
 #' @examples
-#' observation <- ants_L1[[1]]$tables$observation
-#' id <- names(ants_L1)
+#' \dontrun{
+#' # plot ecocomDP formatted dataset
+#' plot_sample_space_time(
+#'   dataset = ants_L1)
+#' 
+#' # plot flattened ecocomDP dataset
+#' plot_sample_space_time(
+#'   flat_data = flatten_dataset(ants_L1))
 #' 
-#' plot_taxa_sample_time(observation, id)
+#' # plot ecocomDP observation table
+#' plot_sample_space_time(
+#'   observation = ants_L1[[1]]$tables$observation)
 #' 
-plot_taxa_sample_time <- function(observation, id, alpha = 1) {
+#' # tidy syntax, filter data by date
+#' ants_L1 %>% 
+#'   flatten_dataset() %>% 
+#'   dplyr::filter(
+#'     lubridate::as_date(datetime) > "2003-07-01") %>%
+#'   plot_sample_space_time(flat_data = .)
+#' 
+#' # tidy syntax, filter data by site ID
+#' ants_L1 %>% 
+#'   flatten_dataset() %>% 
+#'   dplyr::filter(
+#'     as.numeric(location_id) > 4) %>%
+#'   plot_sample_space_time(flat_data = .)
+#' }
+plot_sample_space_time <- function(
```

Comment:
These changes create a backwards incompatibility.

Same comments on `plot_taxa_richness()` apply here.

Reply:
the semantics of the function name didn't make sense. This plotting function plots the distribution of samples or "event_ids" across space/time, not taxa. The name "plot_taxa_sample_time" conveys the idea that we're plotting the number of taxa per sample across time, which is not accurate. We could make an alias to make it backward compatible, but recommend deprecating the old function name. 

I'm open to using a different fxn name than what I've proposed here, but I think the old fxn name is really confusing.

Reply:
Okay, let's go with your recommendation for an alias with deprecation notice.

Reply:
@sokole Add deprecation notice and handling for observation param.

Reply:
@sokole Where's the alias for `plot_taxa_sample_time()`?